pkg/labels/applicator.go

```diff
@@ -54,6 +54,11 @@ func (l Labeled) SameAs(o Labeled) bool {
 	return l.ID == o.ID && l.LabelType == o.LabelType
 }
 
+type WatchResult struct {
+	Labeled []Labeled
+	Valid   bool // Will be false if the underlying watcher has terminated
```

Comment:
Changed back to `[]Labeled`, realized that this safety guarantee can be provided in `select` statements by using `case res, ok := <- resCh`. Adding that to the documentation.